benchmarks/shelley3pools/tx-generator-script.json

```diff
@@ -2,27 +2,33 @@
     { "setInitCooldown": 10 },
     { "setNumberOfInputsPerTx": 1 },
     { "setNumberOfOutputsPerTx": 1 },
-    { "setNumberOfTxs": 500 },
+    { "setNumberOfTxs": 5000 },
     { "setTPSRate": 10 },
     { "setTxAdditionalSize": 0 },
     { "setFee": 0 },
     { "setTTL": 1000000 },
     { "startProtocol": "configuration/configuration-generator.yaml" },
-    { "setEra": "MaryEra" },
+    { "setEra": "AllegraEra" },
     { "setLocalSocket": "logs/sockets/1" },
     { "readSigningKey": "pass-partout", "filePath": "configuration/genesis-shelley/utxo-keys/utxo1.skey" },
     { "secureGenesisFund": "genFund", "genesisKey": "pass-partout", "fundKey": "pass-partout" },
     { "delay": null },
     { "splitFund": [ "fund1", "fund2", "fund3","fund4" ], "sourceFund": "genFund", "newKey": "pass-partout" },
     { "delay": null },
-    { "splitFundToList": "fundList", "sourceFund": "fund1", "newKey": "pass-partout" },
-    { "prepareTxList": "txlist", "newKey": "pass-partout", "fundList": "fundList" },
+    { "splitFundToList": "allegraFunds", "sourceFund": "fund1", "newKey": "pass-partout" },
+    { "prepareTxList": "allegraTxs", "newKey": "pass-partout", "fundList": "allegraFunds" },
+    { "splitFundToList": "maryFunds", "sourceFund": "fund2", "newKey": "pass-partout" },
     {
         "setTargets": [
             { "addr": "127.0.0.1", "port": 3000 },
             { "addr": "127.0.0.1", "port": 3001 },
             { "addr": "127.0.0.1", "port": 3002 }
         ] },
-    { "asyncBenchmark": "threadId", "txList": "txlist" },
-    { "waitBenchmark": "threadId" }
+    { "asyncBenchmark": "allegraThread", "txList": "allegraTxs" },
+    { "waitForEra": "MaryEra" },
+    { "setEra": "MaryEra" },
+    { "prepareTxList": "maryTxs", "newKey": "pass-partout", "fundList": "maryFunds" },
+    { "cancelBenchmark": "allegraThread" },
```

Comment:
Does this mean that we continue creating Allegra transactions while already in Mary era?

Reply:
Yes the `allergraThread` keeps running until it gets canceled by an async exception.
Also `waitForEra` only polls the BS era once a second.
With 7tps there will be ca 3 TXs in the polling delay.